lighthouse/helpers/plates.py

```diff
@@ -529,116 +487,216 @@ def create_cherrypicked_post_body(
     return {"data": {"type": "plates", "attributes": body}}
 
 
-def sample_subjects(samples):
+# WARN - on refactoring, be careful not to lose the distributed functionality where
+# None or empty samples returns None
+def get_source_plates_for_samples(samples):
+    barcodes = get_unique_plate_barcodes(samples)
+    return find_source_plates(query_for_source_plate_uuids(barcodes))
+
+
+def find_source_plates(query: Dict[str, Any]) -> Optional[List[Dict[str, Any]]]:
+    if query is None:
+        return None
+
+    source_plates = app.data.driver.db.source_plates
+
+    source_plate_documents = list(source_plates.find(query))
+
+    logger.info(f"Found {len(source_plate_documents)} source plates")
+
+    return source_plate_documents
+
+
+def get_unique_plate_barcodes(samples):
+    barcodes = set()
+    for sample in samples:
+        barcodes.add(sample[FIELD_PLATE_BARCODE])
+    return list(barcodes)
+
+
+def query_for_source_plate_uuids(barcodes):
+    if barcodes is None or (len(barcodes) == 0):
```

Comment:
I think `if not barcodes...` is more pythonic. I believe it checks for `None` and empty lists.